api/envoy/extensions/filters/network/thrift_proxy/v3/thrift_proxy.proto

```diff
@@ -58,7 +59,7 @@ enum ProtocolType {
   COMPACT = 3;
 
   // The Thrift proxy will use the Thrift "Twitter" protocol implemented by the finagle library.
-  TWITTER = 4;
+  TWITTER = 4 [deprecated = true, (envoy.annotations.deprecated_at_minor_version_enum) = "3.0"];
```

Comment:
From an API perspective, one nit - are we deprecating because we're just not implementing this support or because the Thrift Twitter protocol is fundamentally deprecated in general?

Since xDS is in-principle proxy neutral, this distinction is worth making. That said, in practice, seems good odds that given the complexity here that we're the only implementors of Thrift proxy in practice.

Reply:
> Thrift Twitter protocol is fundamentally deprecated in general

Yes, this, and that it is causing us significant maintenance burden because it depends on a bunch of deprecated PY2 libraries for testing.